### PR TITLE
Add retry logic for AWS throttling errors

### PR DESCRIPTION
**Pull request by molatif-dev**
Add retry logic for AWS throttling errors, and remove TTL for static caches like account_alias, account_id and arn

### CODE REVIEW

File: cs/aws_account/monkies.py [hunk 1]
```diff
@@ -2,9 +2,13 @@
 from threading import RLock
 
 from botocore.loaders import Loader
+import time
```
Comment (faldridge) [inline cs/aws_account/monkies.py:5]: suggestion: `aws_throttling_retry` doesn't belong in the monkies module as it's not a monkey-patch. Consider breaking this out into its own `retry` module.

File: cs/aws_account/monkies.py [hunk 2]
```diff
@@ -35,3 +39,38 @@ def _wrapper(self, *args, **kwargs):
...
+                except Exception as error:
+                    if '(Throttling)' in  str(error):
+                        retries += 1
+                        if retries >= max_retries:
+                            raise error
+                        backoff_seconds = base + (retries * growth_factor)
+                        logger.warning(f"Received AWS throttling error. Retrying in {backoff_seconds} seconds...")
+                        time.sleep(backoff_seconds)
+                    else:
+                        logger.warning(f"Received AWS throttling error. Exhausted all attempts.")
```
Comment (faldridge) [inline cs/aws_account/monkies.py:72]: task: this warning line should be after line 66, I think, when the retries are actually exhausted. Here in line 72, we're warning about throttling when the error received from AWS was not (as best we can tell) a throttling error.

File: cs/aws_account/monkies.py [hunk 3]
```diff
@@ -35,3 +39,38 @@ def _wrapper(self, *args, **kwargs):
...
+
+        """
+    def decorator_retry(func):
+        def wrapper(*args, **kwargs):
+            retries = 0
+            while retries < max_retries:
+                try:
+                    return func(*args, **kwargs)
+                except Exception as error:
+                    if '(Throttling)' in  str(error):
```
Comment (faldridge) [inline cs/aws_account/monkies.py:64]: question: I'd prefer not to compare strings in our AWS error handling. Can we make use of [error codes](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#parsing-error-responses-and-catching-exceptions-from-aws-services) to explicitly check for `LimitExceededException`?

**Review by faldridge — CHANGES_REQUESTED** [review]

File: cs/aws_account/monkies.py (hunk 3 above)
Comment (molatif-dev) [reply cs/aws_account/monkies.py:64]: it's not that straight forward, for two reasons, First of all we wrap every exception to AWSClientException in RegionalAccount._limited()  so I may not have direct access to the original exception error code.
Second of all, throttling is not presented in single error code, at least according to the [boto3 retries doc](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html). The LimitExceededException you shared is specific to Amazon Kinesis, but other services can throw different errors.
For example:
```
Throttling
ThrottlingException
ThrottledException
RequestThrottledException
ProvisionedThroughputExceededException
```
That's why I went with the string approach.

File: cs/aws_account/monkies.py (hunk 1 above)
Comment (molatif-dev) [reply cs/aws_account/monkies.py]: created a new `retry.py`  and moved tje decorator in that file

File: cs/aws_account/monkies.py (hunk 2 above)
Comment (molatif-dev) [reply cs/aws_account/monkies.py]: Good catch. Updated

**Review by faldridge — DISMISSED** [review]
Looks great, thanks!

**Review by faldridge — APPROVED** [review]

**Pull request merged**